docs/madr/decisions/056-meshservice-migration.md

```diff
@@ -0,0 +1,137 @@
+# MeshService migration
+
+* Status: accepted
+
+## Context and Problem Statement
+
+This MADR addresses the switch from not having MeshServices at all to MeshServices
+being automatically created, either from `Services` in Kubernetes or from
+`Dataplanes` in Universal.
+
+When upgrading from 2.8 to 2.9, nothing at the data plane level should change.
+But MeshService behaves differently in several different ways.
+
+### Policy
+
+If for example we have a policy targeted at:
+
+```
+spec:
+  targetRef:
+    kind: MeshService
+    name: demo-app_kuma-demo_svc_5000
+```
+
+when switching over to MeshService, this policy will no longer apply.
```

Comment:
Do we need to put the right use case here?
And also could we add webhook validating here to forbid the old use case and give us the warning to use the latest one?

Reply:
What do you mean by use case? I don't think there's anything to validate. Policies targeting `kuma.io/service` and those targeting `MeshService` objects are both valid